src/parsers/helper.ts

```diff
@@ -53,10 +64,17 @@ export const plugins: ParserPlugin[] = [
 
 export const parseOptions = (filePath: string, strictMode = false): ParserOptions | null => {
   const fileType = supportedFileType(filePath);
+
+  let parserPlugins = plugins;
+  // only add jsx plugin if file is react type.
+  if (fileType && isReactFileType(filePath)) {
```

Comment:
While generally the `x` is used Typescript (though not sure if it's required), 
the overwhelming majority of javascript React files are simply `.js`, not `.jsx`.

`create-react-app` recommended against the `.jsx` extension in 2016. https://github.com/facebook/create-react-app/releases/tag/v0.4.1

Reply:
updated logic to only exclude jsx plugin for ts files.